### PR TITLE
Unit tests to http server

### PR DESCRIPTION
**Pull request by JohanLorenzo**
WIP

### CODE REVIEW

File: src/service_router.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use router::Router;
 use core::marker::Reflect;
 
 
-pub fn create<T: Send + Reflect + ContextTrait + 'static> (context: Shared<T>) -> Router {
+pub fn create<Ctx: Send + Reflect + ContextTrait + 'static> (context: Shared<Ctx>) -> Router {
```
Comment (JohanLorenzo) [inline src/service_router.rs:13]: Topic discussed in #56.

**Comment by JohanLorenzo** [comment]
I'm under the impression this PR causes more problems than it actually solves.

Nonetheless, thanks to it, we noticed:
- 1 bug in stainless
- another in hyper
- 1 missing export in Iron.

This is mainly why we have so many TODOs. I think this PR has still some value, for the sake of documentation. This also paves the road for testing the main routes independently.

What do you think @fabricedesre ? r?

**Comment by fabricedesre** [comment]
Too much noise here for the benefits. Let's postpone until there are less TODO and FIXME.

**Comment by JohanLorenzo** [comment]
Fine by me. I close this PR, until there are less TODOs to deal with.

**Pull request closed without merging**